src/main/java/db/migration/V188__Remove_persist_lineItems_that_should_be_implicit.java

```diff
@@ -0,0 +1,83 @@
+package db.migration;
+
+import org.flywaydb.core.api.migration.jdbc.JdbcMigration;
+
+import java.math.BigDecimal;
+import java.sql.*;
+
+public class V188__Remove_persist_lineItems_that_should_be_implicit implements JdbcMigration {
+
+    @Override
+    public void migrate(Connection connection) throws Exception {
+        // Find all lineItems
+        PreparedStatement psLineItems = connection.prepareStatement("SELECT * FROM `LineItems`");
+
+        try {
+            connection.setAutoCommit(false);
+
+            ResultSet rsLineItems = psLineItems.executeQuery();
+
+            while (rsLineItems.next()) {
+                long lineItemId = rsLineItems.getLong("Id");
+                BigDecimal amount = rsLineItems.getBigDecimal("amount");
+                String description = rsLineItems.getString("description");
+
+                // Should not delete if description doesn't match auto-generated pattern
+                // Look for string 'buyout funds for' or 'work-life balance funds for
+                String buyoutDescription = "Buyout Funds for";
+                String workDescription = "Work-Life Balance Funds for";
+
+                if (description.contains(buyoutDescription) == false && description.contains(workDescription) == false) {
+                    continue;
+                }
+
+                // Should not delete if a cost was set
+                if (amount.compareTo(BigDecimal.ZERO) > 0) {
+                    continue;
+                }
+
+                // Should not delete if comments were made
+                PreparedStatement psComments = connection.prepareStatement("SELECT * FROM `LineItemComments` WHERE `LineItemId` = ?");
+                psComments.setLong(1, lineItemId);
+
+                ResultSet rsComments = psComments.executeQuery();
+                Boolean commentsFound = false;
+
+                while (rsComments.next()) {
+                    commentsFound = true;
```

Comment:
I think a "select count(*)" might be more appropriate instead of looping over the comments.

Reply:
I was fighting with the count syntax for a bit, and went with this as a simpler option. But you're right, the intent is much clearer with count, I'll re-write it.